internal/services/notification/notification_test.go

```diff
@@ -174,6 +178,149 @@ func TestRunWebhookDelivery(t *testing.T) {
 	})
 }
 
+func TestLastRunEventSequence(t *testing.T) {
+	t.Parallel()
+
+	dir := t.TempDir()
+	log := testutil.NewLogger(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ns := setupNotificationService(ctx, t, log, dir)
+	ns.c.WebhookURL = webhookURL
+
+	t.Logf("starting ns")
+
+	runEventsSender := setupRunEventsSender(ctx, t)
+	defer runEventsSender.stop()
+
+	t.Logf("starting run events service")
+
+	ns.runserviceClient = rsclient.NewClient(runEventsSender.exposedURL, "")
+
+	lastRunEventSequenceValue := getLastRunEventSequenceValue(t, ctx, ns)
+	if lastRunEventSequenceValue != 0 {
+		t.Fatalf("expected lastRunEventSequence %d, got: %d", 0, lastRunEventSequenceValue)
+	}
+
+	// test runEventsHandler start from sequence 0
+	runEventsSender.runEvents.addRunEvent(generateRunEvent(1, rstypes.RunPhaseChanged))
+	runEventsSender.runEvents.addRunEvent(generateRunEvent(2, "testrunevents"))
+	expectedLastRunEventSequenceValue := uint64(2)
+
+	err := ns.runEventsHandler(ctx)
+	if err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	lastRunEventSequenceValue = getLastRunEventSequenceValue(t, ctx, ns)
+	if lastRunEventSequenceValue != expectedLastRunEventSequenceValue {
+		t.Fatalf("expected %d last run event got: %d", expectedLastRunEventSequenceValue, lastRunEventSequenceValue)
+	}
+
+	// test new events
+	cleanRunWebhooks(t, ctx, ns)
```

Comment:
I don't understand why you have to remove them from the db. Can't you just use the real expected number?